### PR TITLE
Prepend line directive to copied source files and add test

### PR DESCRIPTION
**Pull request by k1LoW**
Hi @tenntenn and @usatie!

I tried to go ahead with the implementation to append the line directive.

Added a simple test and also responded to @tenntenn's suggestion.

I look forward to gostaticanalysis/testutil being further refined 🚀

### CODE REVIEW

File: mod_test.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+func TestWithModules(t *testing.T) {
+	t.Parallel()
+
+	t.Run("The line directive is appended to the go source codes", func(t *testing.T) {
+		testdata := WithModules(t, analysistest.TestData(), nil)
+		tests := []struct {
+			path string
+			want string
+		}{
+			{filepath.Join(testdata, "src", "a", "a.go"), "//line src/a/a.go:1"},
```
Comment (k1LoW) [inline mod_test.go:20]: The difference from #9  is that the path is not absolute (does not have `/` prefix).

**Comment by k1LoW** [comment]
I finished implementing what I wanted to accomplish with this pull request.

**Comment by k1LoW** [comment]
@tenntenn 
Is there anything else I need to do to merge these changes?

File: testdata/src/a/go.mod [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module a
+
+go 1.21
```
Comment (tenntenn) [inline testdata/src/a/go.mod:3]: go 1.21と書くとgo 1.21rc2が含まれてしまうので、go 1.21.0の方が良いかもです。

**Review by tenntenn — APPROVED** [review]
LGTM

**Pull request merged**